lib-multisrc/greenshit/src/eu/kanade/tachiyomi/multisrc/greenshit/GreenShit.kt

```diff
@@ -226,8 +192,27 @@ abstract class GreenShit(
         }
 
     private fun pageListRequestMobile(chapter: SChapter): Request {
-        val pathSegment = chapter.url.replace("capitulo", "capitulo-app")
-        return GET("$apiUrl$pathSegment", headers)
+        val pathSegment = chapter.url.replace("capitulo", "capitulo-app-token")
+        val newHeaders = headers.newBuilder()
+            .set("x-client-hash", generateToken(scanId, SECRET_KEY))
+            .set("authorization", "Bearer $token")
+            .build()
+        return GET("$apiUrl$pathSegment", newHeaders)
+    }
+
+    private fun generateToken(scanId: Long, secretKey: String): String {
+        val timestamp = System.currentTimeMillis() / 1000
+        val expiration = timestamp + 3600
+
+        val payload = """{"scan_id":$scanId,"timestamp":$timestamp,"exp":$expiration}"""
```

Comment:
Use `buildJsonObject` to separate concerns when it comes to encoding the values.